code/modules/mob/living/living.dm

```diff
@@ -1232,7 +1232,8 @@
 	else
 		mobility_flags &= ~MOBILITY_MOVE
 	var/canstand_involuntary = !HAS_TRAIT(src, TRAIT_FLOORED)
-	var/canstand = canstand_involuntary && !resting
+	var/force_stand = HAS_TRAIT(src, TRAIT_FORCE_STAND)
```

Comment:
Is there any difference between this and being buckled to something with `buckle_lying = 0` ?
Seems like the behavior you are adding already exists.

Reply:
I experimented with this but it felt like using a hammer as screwdriver.

When resisting, you would first unbuckle from the cryo and then resist out of the cryo. So you could unbuckle and lie inside the cryo anyways, not to mention you would need to press two buttons instead of one and all of that.

In the end I think it should be refactored the other way around, so that buckling uses traits as well, as traits are more expansible and easier to understand in this case

Reply:
Agreed, that'd be the best way. Having two different ways of forcing the `lying_angle` is not ideal.
I'm trying to diminish the need of `update_mobility()`, in favor of using traits instead. You can see that on the `TRAIT_IMMOBILIZED` and `TRAIT_FLOORED` I added.
Once all these checks have been reduced to the presence or not of traits we'll be able to remove the need for the proc.

Can your change go in that line, having the changes to `lying_angle` happen based on events, instead of having to compile a big list of requirements first?